src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -666,12 +666,18 @@ public boolean isFilterChangelog() {
     }
 
     /**
-     * Sets the <tt>SVN_REVISION_n</tt> and <tt>SVN_URL_n</tt> environment variables during the build.
+     * Needed to make Mockito happy due to not picking up the parent implementation.
      */
     @Override
-    public void buildEnvVars(AbstractBuild<?, ?> build, Map<String, String> env) {
-        super.buildEnvVars(build, env);
+    public void buildEnvVars(AbstractBuild<?,?> build, Map<String,String> env) {
+        buildEnvironment(build, env);
+    }
 
+    /**
+     * Sets the <tt>SVN_REVISION_n</tt> and <tt>SVN_URL_n</tt> environment variables during the build.
+     */
+    @Override
+    public void buildEnvironment(Run<?, ?> build, Map<String, String> env) {
```

Comment:
Well, one problem is that we'd also have to add a `buildEnvVars(AbstractBuild,Map)` method to the SCMs, since they wouldn't be able to rely on the updated `SCM.buildEnvVars(AbstractBuild,Map)` method...